### ISSUE TITLE
Developer News is not updating

### ISSUE THREAD

**Issue by yannduran** [luminous-software/start-page-plus#25]
From a [Developer Community post](https://developercommunity.visualstudio.com/comments/745465/view.html):

**Sep 24 at 11:19 AM - Fawad Raza**

Hi `@Yann Duran`

Just installed and checked, Developer News is still not refreshed. For me that's all I wanted, rest are fine in new setup.

**Comment by yannduran** [comment]
`@Fawad Raza`

What makes you think it's not refreshing? Do you day-by day you see nothing new? Microsoft often go for several days before they add something. And at least I give you access to all 10 items (they never have more than 10 for some reason), not the just 3 they provide on the installer (which nobody want to go to look for their news there!).

Or do you mean it's not refreshing when you click on refresh?

I uploaded v0.12.1.4 today, did you automatically get the update?

**Comment by yannduran** [comment]
Hi `@Yann Duran`

Yes, Refresh button not working. News are stuck on September 19, 2019.

So I updated the extension as well as Visual Studio today. Perhaps the rss connection between add-in and news is broken in VS 2019 Pro 16.3.0 version.

Tried with _Developer News_ extension. Basically, as you can see below few updates in box, but then if you click the View More News below the box to go to blog, there are updates on webpage that are not showing in either of the windows.

My thinking? Visual Studio 2019 16.3.0 version (released yesterday) broke it.

Page:

Sidebar:

**Comment by yannduran** [comment]
Your _Developer News_ add-in is showing the same items as found in Tools -> Get Tools and Features. But there are newer articles in `https://devblogs.microsoft.com/visualstudio`. So apparently Microsoft's own RSS feed is not synchronized?
Davis, Jon

The RSS feed for devblogs.microsoft.com/visualstudio is `https://devblogs.microsoft.com/visualstudio/feed` and it shows the newer items.

**Comment by yannduran** [comment]
**My reponse:**

I know this can be confusing, especially when nobody's explained how it works. I had to work it out for myself. Basically there are two separate feeds:

  - Developer News
  - Visual Studio Blog

I don't know how many items used to be in the original Developer News feed in VS 2017, because I used to use a start page extension that provided a custom start page which didn't have a Developer News section in it.

I display the exact **same** feed in my _Developer News_ extension that was in _VS 2017's Developer News_, only for some reason that I can't remember, I only display the last **seven** items (I think it was because that was how many items fitted on a page without a scrollbar).

When they killed the start page in VS 2019, and switched to displaying a Developer News section in the VS installer, they displayed a much smaller number of items, and now they only show **three** items in it, relying instead on the View More News link instead. I don't know where the original link used to go to, but now it goes to the VS Blog's page.

The original feed, or a clone of it, is still being maintained by Microsoft, but it only contains a maximum of **ten** items. When they add a new item, it seems that they delete the oldest one at the same time. When I wrote _Start Page+_ I decided to display the full **ten** items.

My _View More News_ link also opens the _Visual Studio Blog_ page, which is what the link in the installer displays.

The items in the two feeds are very similar, but it appears that they may be maintained by two different teams, because they don't always have exactly the same items. I've seen items that were in the Developer News feed that weren't in the VS Blog, and I've seen items that were in the blog that weren't in Developer News.

So the _Refresh_ button hasn't actually stopped working, it seems that you've compared the two feeds at a time where they don't match. The refresh was actually so fast that I could never tell if it had refreshed or not, so I now clear the list first, so there's a visual cue that the refresh has in fact taken place.

I hope all that helps.
